packages/plugin-types/src/backend_identifier.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * These are some utility types to make the BackendIdentifier a bit more self-documenting
+ */
+export type AppId = string;
+export type ProjectName = string;
+export type BranchName = string;
+export type SandboxName = string;
+
+/**
+ * This tuple defines the constituent parts that are used to construct backend stack names
+ * The stack name becomes what identifies a deployed backend.
+ *
+ * To translate to/from a stack name, use the utility methods in @aws-amplify/platform-core
+ */
+export type BackendIdentifier =
+  | {
+      /**
+       * The Amplify AppId for the backend
+       */
+      namespace: Readonly<AppId>;
+      /**
+       * The Amplify branch name for the backend
+       */
+      name: Readonly<BranchName>;
+      /**
+       * Const that determines this BackendIdentifier is for a branch backend
+       */
+      type: Readonly<'branch'>;
```

Comment:
We should use the same type for `BackendIdentifier` and `BackendDeploymentType` as I think they represent the same thing logically.

i.e. just have `BackendType: 'branch' | 'sandbox'` and use it in both places.

Reply:
They do represent the same thing, but changing BackendDeploymentType is going to blow up the PR diff even more. Perhaps that's a moot point now, but if we want to do this I think it should be a follow up.

Reply:
I went ahead and did this. I figured the diff is already large enough that one more rename doesn't really matter. If you think it looks okay, I'll keep it in, otherwise I can revert and we can address this in a follow up.